### PR TITLE
Improve code generation

### PR DESCRIPTION
**Pull request by josetr**
before:

```.cs
internal static global::CppSharp.Parser.Parser __CreateInstance(global::System.IntPtr native, bool skipVTables = false)
{
    return new global::CppSharp.Parser.Parser(native.ToPointer(), skipVTables);
}

get
{
    global::CppSharp.Parser.AST.Expr __result0;
    if (((global::CppSharp.Parser.AST.CaseStmt.__Internal*) __Instance)->lHS == IntPtr.Zero) __result0 = null;
    else if (global::CppSharp.Parser.AST.Expr.NativeToManagedMap.ContainsKey(((global::CppSharp.Parser.AST.CaseStmt.__Internal*) __Instance)->lHS))
        __result0 = (global::CppSharp.Parser.AST.Expr) global::CppSharp.Parser.AST.Expr.NativeToManagedMap[((global::CppSharp.Parser.AST.CaseStmt.__Internal*) __Instance)->lHS];
    else __result0 = global::CppSharp.Parser.AST.Expr.__CreateInstance(((global::CppSharp.Parser.AST.CaseStmt.__Internal*) __Instance)->lHS);
    return __result0;
}
set
{
    ((global::CppSharp.Parser.AST.CaseStmt.__Internal*)__Instance)->lHS = ReferenceEquals(value, null) ? global::System.IntPtr.Zero : value.__Instance;
}
[SuppressUnmanagedCodeSecurity]
[DllImport("CppSharp.CppParser", CallingConvention = global::System.Runtime.InteropServices.CallingConvention.Cdecl,
	EntryPoint="_ZN8CppSharp9CppParser3AST11DefaultStmtC2Ev")]
internal static extern void ctor(global::System.IntPtr __instance);
```

after:

```.cs
internal static Parser __CreateInstance(__IntPtr native, bool skipVTables = false)
{
    return new Parser(native.ToPointer(), skipVTables);
}

get
{
    var __result0 = global::CppSharp.Parser.AST.Expr.__GetOrCreateInstance(((__Internal*)__Instance)->lHS, false);
    return __result0;
}
set
{
    ((__Internal*)__Instance)->lHS = value is null ? __IntPtr.Zero : value.__Instance;
}

[SuppressUnmanagedCodeSecurity, DllImport("CppSharp.CppParser", EntryPoint = "_ZN8CppSharp9CppParser3AST11DefaultStmtC2Ev", CallingConvention = __CallingConvention.Cdecl)]
internal static extern void ctor(__IntPtr __instance);
```

### CODE REVIEW

File: src/Runtime/MarshalUtil.cs [hunk 1]
```diff
@@ -55,5 +56,11 @@ public static IntPtr[] GetIntPtrArray(IntPtr* array, int size)
         {
             return GetArray<IntPtr>(array, size);
         }
+
+        public static T GetDelegate<T>(void*[] vtables, short table, int i) where T : class
+        {
+            var slot = *(void**)((IntPtr)vtables[table] + i * sizeof(IntPtr));
```
Comment (ddobrev) [inline src/Runtime/MarshalUtil.cs:62]: Did we check for null before? I think not.

File: src/CppParser/Bindings/CSharp/i686-pc-win32-msvc/Std.cs [hunk 2]
```diff
@@ -8,6 +8,8 @@
 using System.Runtime.InteropServices;
 using System.Security;
 using System.Runtime.CompilerServices;
+using __CallingConvention = System.Runtime.InteropServices.CallingConvention;
```
Comment (ddobrev) [inline src/CppParser/Bindings/CSharp/i686-pc-win32-msvc/Std.cs:11]: Please add global::.

Comment (josetr) [reply src/CppParser/Bindings/CSharp/i686-pc-win32-msvc/Std.cs:11]: Why is that needed tho?

Comment (ddobrev) [reply src/CppParser/Bindings/CSharp/i686-pc-win32-msvc/Std.cs:11]: To avoid conflicts with generated name-spaces and yes.

Comment (josetr) [reply src/CppParser/Bindings/CSharp/i686-pc-win32-msvc/Std.cs]: Done. For some reason I thought that I could dodge using the global namespace if the statement was at the top of the file :smile:

**Pull request merged**